test/cliargsparser.cpp

```diff
@@ -406,6 +406,46 @@ TEST_CASE("Sets `execute_cmds` if -x/--execute is provided", "[cliargsparser]")
 	}
 }
 
+TEST_CASE("Inserts commands to cmds_to_execute if -x/--execute is provided",
+	"[cliargsparser]")
+{
+	auto check = [](Opts opts, std::initializer_list<std::string> cmds) {
+		CLIArgsParser args(opts.argc(), opts.argv());
+
+		REQUIRE(args.cmds_to_execute.size() == cmds.size());
+
+		for (const auto& cmd : cmds) {
+			const bool found =
+				std::find(args.cmds_to_execute.begin(),
```

Comment:
This checks the presence of commands, but not their order; which is a problem, because command should be executed in the order they're given on the command line.

I apologize, for two things:

1. for not letting you come up with your own solution. Yesterday evening, it seemed a good idea to tell you what I'd do, but now, I think it was a mistake.
2. for forcing a bad, overengineered solution on you. Initializer lists are awesome, but the problem could be solved way easier if we used `std::vector<std::string>` instead. It can be constructed from an initializer list—so callsite would still look nice—and it can be compared with `operator==`, which compares all three things—size, order, and contents.

I feel quite bad about dragging out the review, and even more so since your initial submission was so good. I can merge it as-is and then fix the problems I caused. Do you want me to?

Reply:
>your initial submission was so good
The tests part sucked, and it's good that you pointed it out. :) I wanted to adhere to maintainer's idea of code, but maybe I have should just used vector and stand by it. 

Anyway, how about I change `check` to use vector and just compare the two vectors with `==`?

If not, and those commits are good enough, then I would say go ahead!

Reply:
> maybe I have should just used vector and stand by it

I believe I could use some pushback at times, so yeah—if in doubt, at least make me explain while I believe what I believe :)

> Anyway, how about I change check to use vector and just compare the two vectors with ==?

Go for it! And thank you.